test/unit_tests/unittest_monitoring_frame_serialization_deserialization.cpp

```diff
@@ -15,176 +15,244 @@
 {
-  // Load testdata from dump
-  UDPFrameTestDataWithoutMeasurementsAndIntensities test_data;
-  DynamicSizeRawData serialized_monitoring_frame_message = serialize(test_data.msg_);
+  scanner_udp_datagram_hexdumps::WithoutIntensities without_intensities;
+  DynamicSizeRawData raw = serialize(without_intensities.msg_);
 
-  EXPECT_EQ(test_data.hex_dump.size(), serialized_monitoring_frame_message.size());
+  MonitoringFrameMsg deserialized_msg = deserialize_monitoring_frame(convertToMaxSizeRawData(raw), raw.size());
 
-  for (size_t i = 0; i < test_data.hex_dump.size(); i++)
-  {
-    EXPECT_EQ((uint8_t)serialized_monitoring_frame_message.at(i), test_data.hex_dump.at(i)) << " index " << i;
-  }
+  EXPECT_EQ(without_intensities.msg_, deserialized_msg);
 }
 
-TEST(MonitoringFrameSerializationTest, testSerializationInvariance)
+TEST(MonitoringFrameSerializationTest, shouldSerializeAndDeserializeFrameWithDiagnosticsConsistently)
 {
-  UDPFrameTestDataWithoutIntensities test_data;
-  DynamicSizeRawData raw = serialize(test_data.msg_);
+  scanner_udp_datagram_hexdumps::WithDiagnostics with_diagnostics;
+  DynamicSizeRawData raw = serialize(with_diagnostics.msg_);
 
   MonitoringFrameMsg deserialized_msg = deserialize_monitoring_frame(convertToMaxSizeRawData(raw), raw.size());
 
-  EXPECT_EQ(test_data.msg_, deserialized_msg);
+  EXPECT_EQ(with_diagnostics.msg_, deserialized_msg);
 }
 
-TEST(MonitoringFrameSerializationTest, testSerializationInvariance2)
+TEST(MonitoringFrameSerializationTest, shouldSerializeAndDeserializeSelfConstructedFrameWithDiagnosticsConsistently)
 {
-  MonitoringFrameMsg msg(TenthOfDegree(25), TenthOfDegree(1), 456, { 10, 20, 30, 40 });
+  std::array<std::pair<uint8_t, uint8_t>, 3> byte_bit = { std::make_pair(0, 0),
+                                                          std::make_pair(5, 3),
+                                                          std::make_pair(4, 7) };
+
+  for (const auto& elem : byte_bit)
+  {
+    ASSERT_NE(error_bits.at(elem.first).at(elem.second), Dc::UNUSED)
+        << "The unused diagnostic bits are discarded during deserialization. You should use different test data for "
+           "this test.";
+  }
+
+  MonitoringFrameMsg msg(
+      TenthOfDegree(25),
+      TenthOfDegree(1),
+      456,
+      { 10, 20, 30, 40 },
+      { MonitoringFrameDiagnosticMessage(ScannerId::MASTER, byte_bit.at(0).first, byte_bit.at(0).second),
+        MonitoringFrameDiagnosticMessage(ScannerId::MASTER, byte_bit.at(1).first, byte_bit.at(1).second),
+        MonitoringFrameDiagnosticMessage(ScannerId::SLAVE2, byte_bit.at(2).first, byte_bit.at(2).second) });
 
   DynamicSizeRawData raw = serialize(msg);
 
   MonitoringFrameMsg deserialized_msg = deserialize_monitoring_frame(convertToMaxSizeRawData(raw), raw.size());
 
+  // TODO: The following throws an exception "map::at" if the messages differ. Inspect problem and report to googletest
```

Comment:
Use == if this works

Reply:
Please resolve first https://github.com/PilzDE/psen_scan_v2/pull/54#discussion_r508343668 Then we should be able to remove the comment and use `EXPECT_EQ`.